File: app/src/main/java/com/tammamkhalaf/myuaeguide/databases/hereDeveloper/PlacesDao.java
package com.tammamkhalaf.myuaeguide.databases.hereDeveloper;

import androidx.room.Dao;
import androidx.room.Delete;
import androidx.room.Insert;
import androidx.room.Query;

import com.tammamkhalaf.myuaeguide.categories.hereDeveloper.discoverHere.LockupEntrypoint.PlaceResponse;

import java.util.List;

import io.reactivex.Completable;
import io.reactivex.Single;

@Dao
public interface PlacesDao {

    @Insert
    Completable insertPlace(PlaceResponse response);

    @Query("select * from placestable")
    Single<List<PlaceResponse>> getPlaces();

    @Delete
    Completable delete(PlaceResponse place);

}

File: shared/src/commonMain/kotlin/com/tammamkhalaf/kmmsharedmodule/databases/hereDeveloper/PlacesDao.kt
package com.tammamkhalaf.myuaeguide.databases.hereDeveloper

import androidx.room.Dao

@Dao
interface PlacesDao {
    @Insert
    fun insertPlace(response: PlaceResponse?): Completable?

    @Query("select * from placestable")
    fun getPlaces(): Single<List<PlaceResponse?>?>?

    @Delete
    fun delete(place: PlaceResponse?): Completable?
}